### ISSUE TITLE
If user checks out a tag in an sub module -- have git-fleximod handle it as that sub module is out of sync

### ISSUE THREAD

**Issue by ekluzek** [ESMCI/git-fleximod#38]
I'm see the following behavior in ctsm5.2.007:

```
cd src/fates
git checkout sci.1.76.0_api.35.0.0
cd ../..
./bin/fleximod status
                 fates at hash sci.1.76.0_api.35.0.0
                  cism at tag cismwrap_2_2_001
                   rtm at tag rtm1_0_79
                mosart at tag mosart1_0_49
             mizuRoute at tag cesm-coupling.n02_v2.1.2
            ccs_config at tag ccs_config_cesm0.0.106
                  cime at tag cime6.0.246
                 cmeps at tag cmeps0.14.63
                 cdeps at tag cdeps1.0.34
                 share at tag share1.0.19
                   mct at tag MCT_2.11.0
            parallelio at tag pio2_6_2
           doc-builder at tag v1.0.8
```

NOTE, that the only clue that FATES is out of sync is the use of the word "hash" rather than "tag". And because it doesn't **see** it as out sync, it won't update it when git fleximod is run.

However, running this does restore it...

`git submodule update --init --checkout --recursive`

**Comment by ekluzek** [comment]
Also note that the behavior for either a branch or a hash DOES note it as out of sync and will update it on git-fleximod update. So I'd just like the same behavior for a tag as well.

**Comment by jedwards4b** [comment]
Please indicate how you arrived at this error.   The description above doesn't do it for me.

**Comment by ekluzek** [comment]
Ahh, I left out the critical step. So I edited the above to include the checkout to a tag

`git checkout sci.1.76.0_api.35.0.0`

try that...

**Comment by jedwards4b** [comment]
Fixed by #40

**Comment by ekluzek** [comment]
This works for me so closing
